common/src/node.rs

```diff
@@ -28,6 +28,33 @@ impl ChunkId {
 }
 
 impl Graph {
+    /// Returns the NodeState for the given node, panicking if it isn't initialized.
+    #[inline]
+    pub fn node_state(&self, node_id: NodeId) -> &NodeState {
+        self[node_id].state.as_ref().unwrap()
+    }
+
+    /// Initializes the NodeState for the given node and all its ancestors if not
+    /// already initialized.
+    pub fn ensure_node_state(&mut self, node_id: NodeId) {
```

Comment:
We often seem to need to access the state immediately after ensuring it. Would that code be more concise if this function returned the `&mut NodeState` directly?

Reply:
It should only need to return a `&NodeState`, but otherwise, that seems like a good suggestion to me.